sanic_openapi/autodoc.py

```diff
@@ -0,0 +1,93 @@
+import inspect
+import warnings
+
+import yaml
+
+
+class OpenAPIDocstringParser(object):
+    def __init__(self, docstring):
+        """
+        Args:
+            docstring (str): docstring of function to be parsed
+        """
+        if docstring is None:
+            docstring = ""
+        self.docstring = inspect.cleandoc(docstring)
+
+    def to_openAPI_2(self):
+        """
+        Returns:
+            json style dict: dict to be read for the path by swagger 2.0 UI
+        """
+        raise NotImplementedError()
+
+    def to_openAPI_3(self):
+        """
+        Returns:
+            json style dict: dict to be read for the path by swagger 3.0.0 UI
+        """
+        raise NotImplementedError()
+
+
+class YamlStyleParametersParser(OpenAPIDocstringParser):
+    def _parse_no_yaml(self, doc):
+        """
+        Args:
```

Comment:
Since sanic-openapi is available only for Python 3.5 and upper and since Python 3.5 includes native type hinting why don't use it? Ex:
```python
def _parse_no_yaml(self, doc: str) -> dict:
    """
    Awesome doc string

    :param t: section of doc before yaml, or full section of doc
    :return: json style dict: dict to be read for the path by swagger UI
    """
```

Reply:
Sure, added